azure-quantum/azure/quantum/qiskit/provider.py

```diff
@@ -120,18 +120,31 @@ def backends(self, name=None, **kwargs):
             backend_list, filters=workspace_allowed, **kwargs
         )
 
-        # Also filter out non-default backends.
-        default_backends = list(
-            filter(
-                lambda backend: self._match_all(
-                    backend.configuration().to_dict(), {"is_default": True}
-                ),
-                filtered_backends,
-            )
-        ) 
-       # If default backends were found - return them, otherwise return the filtered_backends collection.
-       # The latter case could happen where there's no default backend defined for the specified target.  
-        if len(default_backends) > 0:
+        def filter_non_passthrough_defaults(filtered_backends):
+            filtered_defaults = {}
+
+            for backend in filtered_backends:
+                config = backend.configuration().azure
+                is_default =  config["is_default"]
+                is_passthrough = config["is_passthrough"]
+                name = backend.name()
+
+                # Process only if the backend is marked as default
+                if is_default:
+                    # If the name is not in the dictionary, add the backend
+                    # If the current backend is not passthrough, update it in the dictionary
+                    if name not in filtered_defaults or not is_passthrough:
```

Comment:
what is the meaning of `is_passthrough` here and why do we need to update the `filtered_defaults` if we found another backend with the same name that has `is_passthrough = False`?

Reply:
To give a bit more context, now that we have qiskit QIR by default, IonQ passthrough backends are harder to get (as if someone specifies something like `get_backend("ionq.simulator", input_data_format="ionq.circuit.v1")` this will actually still fail (as it matches to both the native and qis gateset now, as qis is no longer the default out of the two). So, this new flag allows for easier fetching of passthrough backends, where we set the default passthroguh and QIR backends to `is_default=True`, and then if we've matched to both backends (the default QIR and passthrough backend), we'll always prefer to use the `is_passthrough=False` backend, aka the QIR backend.

TLDR: is_passthrough is a new flag that basically lets us to still have defaults between passthrough backends, while also maintaining QIR as our main default.